### PR TITLE
refactor: unit test and docstring FilterBadges

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -** 
Cleans up FilterBadges component and adds docstrings and unit tests.

FilterBadge configs now go into `constants/filters`, alongside a StrictDict of filter names.
Configs include: `{ displayName, connectedFilters, filterOrder?, hideValue? }`.
This includes all values necessary to populate a Filter Badge minus:
* value - now configured in a per-key selector object in root selector module
* handleBadgeClose method - tied to queryParams, which are managed by Gradebook component, and must be passed down.

filterBadgeConfig selector takes a filterName and returns the full set of props (minus handleBadgeClose) necessary for populating the filter badge.

Based on this, SingleValue and Range type FilterBadge components are no longer necessary, and the FilterBadge component is much simpler.

Final FilterBadges component is now trivially small.

Partial ticket completion
JIRA: [AU-3](https://openedx.atlassian.net/browse/AU-3)

**What changed?**

- [ More in depth breakdown of changes ]
- [ Peripheral things that got changed ]
- [ etc... ]

**Developer Checklist**
- [x] Test suites passing
- [ ] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x] I've done a visual code review
- [x] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: src/data/selectors/filters.test.js [hunk 1]
```diff
@@ -74,6 +75,17 @@ describe('filters selectors', () => {
     });
   });
 
+  describe('isDefault', () => {
+    it('returns true iff value is equal to initialFilters[name]', () => {
```
Comment (leangseu-edx) [inline src/data/selectors/filters.test.js:79]: typo

File: src/components/FilterBadges/index.jsx [hunk 2]
```diff
@@ -1,123 +1,25 @@
-import { connect } from 'react-redux';
-
+/* eslint-disable import/no-named-as-default */
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import initialFilters from 'data/constants/filters';
-import selectors from 'data/selectors';
+import { badgeOrder } from 'data/constants/filters';
```
Comment (leangseu-edx) [inline src/components/FilterBadges/index.jsx:5]: 👍

File: src/data/selectors/filters.test.js (hunk 1 above)
Comment (muselesscreator) [reply src/data/selectors/filters.test.js:79]: iff = "if and only if"

**Review by leangseu-edx — APPROVED** [review]

**Review by mattcarter — APPROVED** [review]
approved.  Smoke tests look good 👍

**Review by nsprenkle — APPROVED** [review]
Visual review looks good but have manually verified behavior.

**Review by jnlapierre — APPROVED** [review]
LGTM

File: src/components/FilterBadges/FilterBadge.jsx [hunk 3]
```diff
@@ -1,47 +1,65 @@
...
  * FilterBadge
  * Base filter badge component, that displays a name and a close button.
  * If showValue is true, it will also display the included value.
- * @param {string} name - filter name
- * @param {bool/string} value - filter value
- * @param {func} onClick - close/dismiss filter event
- * @param {bool} showValue - should the Value be displayed instead of just name?
+ * @param {func} handleClose - close/dismiss filter event, taking a list of filternames
+ *   to reset when the filter badge closes.
+ * @param {string} filterName - api filter name (for redux connector)
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/components/FilterBadges/FilterBadge.jsx:15]: no.  Is still there, and necessary, just not passed through to the final inner component.
I always put the things that need to be passed through from the parent in the doc-string, not what is consumed by the final component, and specifically never things from redux.
